pkg/grafana/grafana.go

```diff
@@ -0,0 +1,156 @@
+package grafana
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/models"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	SharedOrgName = "Shared Org"
+)
+
+func CreateOrganization(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organization Organization) (Organization, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("creating organization")
+	err := assertNameIsAvailable(ctx, grafanaAPI, organization)
+	if err != nil {
+		return organization, errors.WithStack(err)
+	}
```

Comment:
Why do we need to check for name availability first? I would rather that we execute the create request and handle the not found case here

Reply:
Asserting name is available and not found are totally different use cases.

Reply:
The goal here is to validate early to avoid weird errors later on